### ISSUE TITLE
Problem deleting block volume on Openstack

### ISSUE THREAD

**Issue by elliottsam** [hashicorp/terraform#4110]
I am seeing the following error when trying to remove a secondary volume that has been added, running Terraform 0.6.6, it will eventually complete successfully, but this can take 20-30mins.

```
2 error(s) occurred:

* openstack_blockstorage_volume_v1.ms.1: Error deleting OpenStack volume: Expected HTTP response code [202 204] when accessing [DELETE ...], but got 400 instead
{"badRequest": {"message": "Invalid volume: Volume status must be available or error, but current status is: deleting", "code": 400}}
* openstack_blockstorage_volume_v1.ms.2: Error deleting OpenStack volume: Expected HTTP response code [202 204] when accessing [DELETE ...], but got 400 instead
{"badRequest": {"message": "Invalid volume: Volume status must be available or error, but current status is: deleting", "code": 400}}
```

**Comment by jtopjian** [comment]
Sorry for the delay with this. In order for this error to occur, the volume must already be in a state of `deleting` at the time the Delete request happened, which is odd. 

Do you have an example Terraform configuration that can reproduce this? Is there another way that can trigger or reproduce this situation?

**Comment by jtopjian** [comment]
Coincidentally enough, I discovered how to create this issue by working on #4090. I have a fix for this, and since the fix is part of the fix for #4090, I'll create a single PR to resolve both that issue and this.

**Comment by jtopjian** [comment]
@elliottsam I decided to open a PR that targets this issue specifically. #4204 should fix this. If you're able to test it out, let me know how it works.
